vnext/src/Libraries/Text/Text.uwp.js

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
```

Comment:
>Copyright (c) Facebook, Inc. and its affiliates. [](start = 3, length = 48)

I'd like to avoid forking code if possible.  Have you explored doing this optimization only in the native code?  

My understanding is that we use RawTextViewManager to handle the text content by creating a Run for it, and then TextViewManager ends up adding the Run as a child in TextViewManager::AddView().  Could we change this to optimize away the InlineCollection when possible?  As a starting point, I'm wondering if we could keep the code in RawTextViewManager the same, but change TextViewManager::AddView() to detect the first Run is being added, and grab the Text out of the Run and set it as the Text property.

Reply:
I thought of this approach initially and talked to @stecrain about it but soon realized that this would require extra processes since UIManagerModule will have to call CreateView for every RawText  and SetChildren for every TextBlock regardless if they are needed. Assuming most users won't be using nested TextBlocks, it would save time to do this in the JS side.

Reply:
Sure it would save time, but it requires a fork to the code which I'd really like to avoid.  If using fast-path saves us, say 80% of the total cost, even with the extra view being created, I'd prefer that solution

Reply:
We're revisiting a native solution. Thank you for taking a look!
You're correct that fast path makes such a dramatic difference that creating the extra inline object is negligible, just wanted to go for max performance! 🚀